### PR TITLE
docs: add computeWaterfallData utility for waterfall charts

### PR DESCRIPTION
**Pull request by MaxGhenis**
## Summary

Adds a `computeWaterfallData()` utility function that transforms raw data into range values for rendering waterfall charts with the existing `<Bar>` component.

- **New file**: `src/util/waterfall.ts` — pure function with full TypeScript types
- **Export**: `computeWaterfallData`, `WaterfallOptions`, `WaterfallDatum` from package index
- **Tests**: 12 unit tests covering positive/negative/mixed values, total bars, accessor functions, edge cases
- **Storybook**: Example story demonstrating a profit & loss waterfall

## Screenshot

Simple waterfall

## Motivation

Waterfall charts are commonly needed in financial, analytical, and reporting dashboards. Currently users must manually compute bar positions. This utility does that computation, outputting `[barBottom, barTop]` range values that Recharts `<Bar>` already supports natively.

### Usage

```tsx
import { BarChart, Bar, XAxis, YAxis, Cell, computeWaterfallData } from "recharts";

const raw = [
  { name: "Revenue", value: 100 },
  { name: "Costs", value: -30 },
  { name: "Profit", value: 70, isTotal: true },
];

const data = computeWaterfallData({ data: raw, dataKey: "value", totalKey: "isTotal" });

<BarChart data={data}>
  <XAxis dataKey="name" />
  <YAxis />
  <Bar dataKey="waterfallRange">
    {data.map((d, i) => (
      <Cell key={i} fill={d.value >= 0 ? "#00897B" : "#757575"} />
    ))}
  </Bar>
</BarChart>
```

## Design decisions

- **Pure utility, not a new component** — keeps the API surface minimal and composable
- **Leverages existing range bar support** — no new rendering code needed
- **Supports accessor functions** — for `dataKey` and `totalKey`, matching Recharts conventions
- **Spreads original data** — output objects include all original fields for easy use with XAxis, tooltips, etc.

## Test plan

- [x] 12 unit tests pass
- [x] Build output snapshots updated
- [x] Storybook story renders correctly (see screenshot above)
- [ ] CI passes

Related: #7010

## Summary by CodeRabbit

* **New Features**
  * Introduced a new Waterfall chart example in the documentation that visualizes financial metrics including revenue, service fees, fixed and variable operating costs, tax expenses, and final profit. Features color-coded representation (green for positive, red for negative, blue for summary totals) and interactive tooltips for detailed insights.

### CODE REVIEW

File: storybook/stories/Examples/BarChart/WaterfallChart.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        <YAxis tickFormatter={(v: number) => `$${v.toFixed(1)}bn`} />
+        <Tooltip
+          formatter={(_value: [number, number], _name: string, props: { payload: (typeof budgetWaterfall)[0] }) => [
+            `$${props.payload.value.toFixed(1)}bn`,
+            props.payload.name,
+          ]}
+        />
+        <Bar dataKey="waterfallRange" isAnimationActive={false}>
+          {budgetWaterfall.map((entry, index) => (
+            <Cell key={`cell-${index}`} fill={getFill(entry)} />
```
Comment (PavelVanecek) [inline storybook/stories/Examples/BarChart/WaterfallChart.stories.tsx]: Please don't use `Cell` in new charts, we're going to remove it in 4.0

**Comment by ckifer** [comment]
This kind of feels like user land code to me as a utility. Not that it's a bad thing, but not all waterfall implementations may want this same behavior. Once we export this we are responsible for it in a way that we are not if we just say, document it as a possible option

**Comment by PavelVanecek** [comment]
+1 to @ckifer. I think this would be a great example chart (as a sibling of https://recharts.github.io/en-US/examples/Candlestick/ and https://recharts.github.io/en-US/examples/PopulationPyramid/) but not as a general library export. It works for one specific data format but not others, we're not helping.

Would you like to turn this into a website example?

**Comment by PavelVanecek** [comment]
Alternatively we could start a second project, one where we're more opinionated about the data shapes, and provide some pre-made charts. Sort of competition to what shadcn and evilcharts are doing. If we put it in separate npm package then we have 100% freedom, if we put it in the main lib we need to double check that we can tree-shake the extra code.

**Comment by MaxGhenis** [comment]
Makes sense — I'll convert this to a website example (like Candlestick/PopulationPyramid) rather than a library export. Will update the PR.

**Comment by MaxGhenis** [comment]
Updated — removed the library export entirely and converted to a self-contained website example at `www/src/docs/exampleComponents/BarChart/WaterfallExample.tsx` (following the Candlestick/PopulationPyramid pattern). The waterfall computation logic is inline in the example. All unit tests pass.

**Review by PavelVanecek — APPROVED** [review]
A new VR test for the new example would be nice too
